components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/cli/BuildTimeMetricsFields.java

```diff
@@ -0,0 +1,48 @@
+package com.gradle.enterprise.cli;
+
+import com.gradle.enterprise.model.BuildTimeMetrics;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+public enum BuildTimeMetricsFields {
+    // The order the enums are defined controls the order the fields are printed in the CSV
+    INITIAL_BUILD_TIME("Initial Build Time", d -> formatDuration(d.getInitialBuildTime())),
+    INSTANT_SAVINGS("Instant Savings", d -> formatDuration(d.getInstantSavings())),
+    INSTANT_SAVINGS_BUILD_TIME("Instant Savings Build Time", d -> formatDuration(d.getInstantSavingsBuildTime())),
+    PENDING_SAVINGS("Pending Savings", d -> formatDuration(d.getPendingSavings())),
+    PENDING_SAVINGS_BUILD_TIME("Pending Savings Build Time", d -> formatDuration(d.getPendingSavingsBuildTime())),
+    ;
+
+    public final String label;
+    public final Function<BuildTimeMetrics, String> value;
+
+    BuildTimeMetricsFields(String label, Function<BuildTimeMetrics, String> value) {
+        this.label = label;
+        this.value = value;
+    }
+
+    public static Stream<BuildTimeMetricsFields> ordered() {
+        return Arrays.stream(BuildTimeMetricsFields.values());
+    }
+
+    private static String formatDuration(Duration duration) {
```

Comment:
Isn't there a formatter built-in to Java to render durations?

A quick search reveals nothing. But I don't think you need to do any math:

System.out.println(String.format("%sh %sm %ss", 
                d.toHoursPart(), 
                d.toMinutesPart(), 
                d.toSecondsPart()));

Reply:
Got it.